src/javascript/app/pages/user/view_popup/view_popup.js

```diff
@@ -767,6 +767,9 @@ const ViewPopup = (() => {
         if (response.error) {
             if (response.error.code !== 'AlreadySubscribed' && response.echo_req.contract_id === contract_id) {
                 showErrorPopup(response, response.error.message);
+                if (response.error.code === 'InvalidToken') {
```

Comment:
I think it should be more general, can't we check this in socket file for response of any call and refresh it there?